### DISCUSSION TITLE
Run testcase and after that execute testcases that validate result

### DISCUSSION THREAD

**Discussion by Stefanhg** [pytest-dev/pytest#11318] [Q&A]
Hello,

I got a testcase where first you execute a one hour test and afterwards validate the results in multiple steps. The testdata comes from the same loop of the testcase, just different aspects of the testdata being validated in each validation step.
Visually that shows a single testcase but I think it could be improved. 
I am unsure if the way that I have implemented is a nice-to-do way or It could be done in a more clean Pytest-ish way. 
A class in Pytest is seen as a testsuite right? So if the testsuite itself executed the test, that would be incorrect? Or would that be acceptable?
Any better way of doing this?

Example of the concept:
```
import pytest

@pytest.fixture()
def testcase_fixture():
    if not hasattr(pytest, "long_test_results"):
        print("EXECUTING TEST")
        long_test_with_multiple_results()
    yield ""

def long_test_with_multiple_results():
    # Testcase
    pytest.long_test_results = [True, False]


@pytest.mark.usefixtures("testcase_fixture")
class TestLongTest:

    def test_verdict_long_test_1(self):
        assert pytest.long_test_results[0]

    def test_verdict_long_test_2(self):
        assert pytest.long_test_results[1]
```

**Comment by RonnyPfannschmidt** [comment]
the subtests plugin allows to split tests into multiple steps with descriptions

**Reply by Stefanhg** [reply]
I just tested it.
It does meet the criteria that i specified, but in terms of how it represents itself I am not a fan.
It needs some way of visually separating each subtest, for example an argument.

I came up with another solution. It has some obvious flaws like one test has to be executed before the data is available.
If using the pytest-ordering plugin you could have the main testcase and then the 2 verdict testcases afterwards, then set the priority of the actual test as top and the others as lowest.
The flaw could surely be resolved by making a fixture that checks if the 'test_long_test' does not exists in the request.session.items fixture and then skip the test.

```
class TestLongTest:
    test_result = list()

    @pytest.mark.first
    def test_long_test(self):
        TestLongTest.test_result = [True, False]

    @pytest.mark.last
    def test_verdict_long_test_1(self):
        assert TestLongTest.test_result[0]

    @pytest.mark.last
    def test_verdict_long_test_2(self):
        assert TestLongTest.test_result[1]
```
